InventoryLowQuantityNotification/Model/ResourceModel/LowQuantityCollection.php

```diff
@@ -300,11 +302,68 @@ private function addEnabledSourceFilter(): void
     private function addSourceItemInStockFilter(): void
     {
         $condition = '(' . SourceItemInterface::QUANTITY . ' > 0 AND main_table.status = ' .
-            SourceItemInterface::STATUS_IN_STOCK . ') OR
-            (' . SourceItemInterface::QUANTITY . ' = 0)';
+            SourceItemInterface::STATUS_IN_STOCK . ')';
         $this->getSelect()->where($condition);
     }
 
+    /**
+     * Filter collection based on notifyStockQty
+     *
+     * @param int|null $storeId
+     * @return void
+     */
+    public function useNotifyStockQtyFilter(int $storeId = null): void
```

Comment:
Is it possible to add filters to SQL query before execution? It's not optimal to get all DB data and filter it on the business logic side. 
Also, it would be better to extract logic in the separate class and avoid new public method creation.

Reply:
It is difficult to put it in the SQL query before execution. As you see, some conditions depend on the config settings as well as sourceItem based config settings. Separating and combining two logic is difficult to be in the same query. Rather, I will introduce a new helper class to filter the products based on the quantity.

Reply:
Created a new helper class